umcontroller/umcontroller.go

```diff
@@ -1021,6 +1041,51 @@ func (umCtrl *Controller) updateComplete(ctx context.Context, e *fsm.Event) {
 	umCtrl.cleanupCurrentComponentStatus()
 }
 
+func (umCtrl *Controller) updateNodeInfo(nodeInfoProvider NodeInfoProvider) error {
+	ids, err := nodeInfoProvider.GetAllNodeIDs()
+	if err != nil {
+		return aoserrors.Wrap(err)
+	}
+
+	for _, id := range ids {
+		nodeInfo, err := nodeInfoProvider.GetNodeInfo(id)
+		if err != nil {
+			return aoserrors.Wrap(err)
+		}
+
+		if nodeInfo.Status == "unprovisioned" {
+			continue
+		}
+
+		umCtrl.connections = append(umCtrl.connections, umConnection{
```

Comment:
We should add only those nodes that have UM Aos component.

Reply:
fixed